### ISSUE TITLE
list_blobs does not work if prefix is empty string

### ISSUE THREAD

**Issue by rongduan-zhu** [Azure/azure-storage-python#451]
### Which service(blob, file, queue) does this issue concern?
blob

### What problem was encountered?
When calling `BlockBlobService.list_blobs(prefix="")` with `prefix=""` will cause an authentication error to be thrown thrown.

```
azure.common.AzureHttpError: Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.ErrorCode: AuthenticationFailed
<?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:d7eda6cf-b01e-0024-55e6-e0d309000000
Time:2018-05-01T00:52:44.4779222Z</Message><AuthenticationErrorDetail>The MAC signature found in the HTTP request '/LACPvjjxGLO/jhbYW19z/4xlPYlZt9DYMp9Rvb/H88=' is not the same as any computed signature. Server used following string to sign: 'GET











x-ms-client-request-id:f4aba85a-4cd9-11e8-b3d1-645106571c89
x-ms-date:Tue, 01 May 2018 00:52:43 GMT
x-ms-version:2017-07-29
/rozhdevv2/test
comp:list
prefix:
restype:container'.</AuthenticationErrorDetail></Error>
```

### Have you found a mitigation/solution?
Yes, the reason is that empty prefix is omitted as part of signing process but since it is included in the query param strings sent to the server, the server is actually verifying the signing with prefix set as empty string, which causes the signatures to mismatch.

Note: for table service, please post the issue here instead: https://github.com/Azure/azure-cosmosdb-python.

**Comment by rongduan-zhu** [comment]
After further investigation, this is caused by calling `list_blobs` with `prefix=""`. Updated the title and description to reflect this and proposed #452 as a fix.

**Comment by zezha-msft** [comment]
@rongduan-zhu Thanks for the PR, I'll keep this issue open until the fix is released.

**Comment by zezha-msft** [comment]
The fix is in 1.2.0rc1. I'll close this issue once the preview period of 1.2.0 is over.

**Comment by zezha-msft** [comment]
Published in version 1.3.0.
